### ISSUE TITLE
Groupadd error when group already exists

### ISSUE THREAD

**Issue by fj-morales** [irods/irods#7057]
Hi, 

I'm trying to install irods in an LDAP environment where I need to use a given user and group accounts. 

The following line fails when the group already exists:

`cmd = ['groupadd', '-r', irods_group]`

https://github.com/irods/irods/blob/25d8f67df6c123ee6f0b3fe4695e4042b52cdb59/scripts/setup_irods.py#L264

Currently, I'm circumventing the issue by adding the flag -f to the command like this 
` cmd = ['groupadd', '-fr', irods_group] `

Could this be used as the default behaviour?

Thanks!

**Comment by trel** [comment]
This line just above is supposed to detect if the group already exists...

https://github.com/irods/irods/blob/25d8f67df6c123ee6f0b3fe4695e4042b52cdb59/scripts/setup_irods.py#L262

Can we make that 'check' a bit smarter?   It's not able to 'see' the LDAP groups, I think?

**Comment by korydraughn** [comment]
I don't think we can make that the default. Some admins may prefer the current behavior.

A safer path forward would be to add a new option that instructs setup to include the `-f` flag. That gives admins the choice to opt into it or not.

**Comment by trel** [comment]
Agreed - can't add the `-f` by default.    How can we make it see/know about the LDAP groups?

**Comment by trel** [comment]
I thought this was the job of the OS... if the LDAP groups aren't visible to the OS... then... well, you're right, we have to provide another wrapper / layer... or we allow them to 'turn this part off'?  `--skip-service-account-group-creation` ?

**Comment by korydraughn** [comment]
> I thought this was the job of the OS

I'm not sure how LDAP integrates with the OS.

> `--skip-service-account-group-creation`?

Do you know why setup creates the irods group? Is it required by anything?

We should investigate that. If it isn't used, then maybe we remove that logic.

**Comment by trel** [comment]
we create the unix account and we also need a group for it to belong to… 

yes maybe we can revisit that process/need/assumption from first principles.

**Comment by trel** [comment]
related to https://github.com/irods/irods/issues/6246 - merged late march 2022.
